### PR TITLE
Socket2 for freertos

### PR DESCRIPTION
**Pull request by DavidHorton5339**
Added FreeRTOS hooks for socket2, using bindings in lwip crate.

### CODE REVIEW

File: src/sys/freertos.rs [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+}
+
+impl FromRawSocket for crate::Socket {
+    unsafe fn from_raw_socket(socket: RawSocket) -> crate::Socket {
+        crate::Socket::from_raw(socket as Socket)
+    }
+}
+
+#[test]
+fn in_addr_convertion() {
```
Comment (maharg101) [inline src/sys/freertos.rs:702]:
```suggestion
fn in_addr_conversion() {
```

File: src/sys/freertos.rs [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+
+    let ip = Ipv4Addr::new(127, 34, 4, 12);
+    let raw = to_in_addr(&ip);
+    let a = raw.s_addr;
+    assert_eq!(a, u32::from_ne_bytes([127, 34, 4, 12]));
+    assert_eq!(from_in_addr(raw), ip);
+}
+
+#[test]
+fn in6_addr_convertion() {
```
Comment (maharg101) [inline src/sys/freertos.rs:718]:
```suggestion
fn in6_addr_conversion() {
```

**Review by maharg101 — APPROVED** [review]
Not sure whether a couple of test names have typos carried over from other platforms ? It might be simpler to leave as-is, if so. Happy for this to go in either way.

File: src/sys/freertos.rs (hunk 1 above)
Comment (DavidHorton5339) [reply src/sys/freertos.rs:702]: Typo was indeed carried over from other platforms. I think I'll leave as it is.

**Pull request merged**